portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -109,4 +109,11 @@ private String convertToJsonUsingGson(List<CommentsClass> comments) {
     String json = gson.toJson(comments);
     return json;
   }
+  public String getJsonString(List<CommentsClass> comments){
```

Comment:
Does this need to be public? Where would it be used?

Reply:
In order to test the convertToJsonUsingGson which is private I set up a getter function which is public so that i can test it

Reply:
Adding a function just for testing isn't a good idea. Instead what we can do is test the public API where convertToJsonUsingGson() is actually being used. 
Further read: https://en.wikipedia.org/wiki/Code_coverage.

Basically assume I have 2 functions A() and B() where A is public and B is private. B is being called within A.
What I want to do is write a test for A() such that the code for B is covered by the test. If every step of B is covered then we're good.

Generally we should only write tests for our public interface.